packages/ingest/granule.js

```diff
@@ -574,6 +578,131 @@ function selector(type, protocol) {
   throw new Error(`${type} is not supported`);
 }
 
+/**
+ * Gets metadata for a cmr xml file from s3
+ *
+ * @param {string} xmlFilePath - S3 URI to the xml metadata document
+ * @returns {string} returns stringified xml document downloaded from S3
+ */
+async function getMetadata(xmlFilePath) {
+  if (!xmlFilePath) {
+    throw new XmlMetaFileNotFound('XML Metadata file not provided');
+  }
+
+  // GET the metadata text
+  // Currently, only supports files that are stored on S3
+  const parts = xmlFilePath.match(/^s3:\/\/(.+?)\/(.+)$/);
+  const obj = await getS3Object(parts[1], parts[2]);
+  return obj.Body.toString();
+}
+
+/**
+ * Parse an xml string
+ *
+ * @param {string} xml - xml to parse
+ * @returns {Promise<Object>} promise resolves to object version of the xml
+ */
+async function parseXmlString(xml) {
+  return new Promise((resolve, reject) => {
+    xml2js.parseString(xml, xmlParseOptions, (err, data) => {
+      if (err) return reject(err);
+      return resolve(data);
+    });
+  });
+}
+
+async function postS3Object(destination, options) {
+  await promiseS3Upload(
+    { Bucket: destination.bucket, Key: destination.key, Body: destination.body }
+  );
+  if (options) {
+    const s3 = aws.s3();
+    s3.deleteObject(options).promise();
```

Comment:
you should either return this or use `await`. If you decide to return make sure the function always return something.

Reply:
changed